### ISSUE TITLE
IBAN not accepted

### ISSUE THREAD

**Issue by TheNetStriker** [schoero/SwissQRBill#215]
My IBAN ends with the letter "K", but this is not accepted by this library. I verified the IBAN on various sites (e.g. https://rechnerli.ch/iban-pruefen-validieren) and it is valid.

Could you please add support for IBAN with letters at the end?

**Comment by schoero** [comment]
SwissQRBill uses the npm module [iban](https://www.npmjs.com/package/iban) to verify the provided IBAN.

Unlike the web page you have mentioned, the module verifies not only if the checksum is valid, but also whether the structure of the IBAN matches country specific specifications. Based on the source code of the iban module, letters in the account number should be supported. Therefore I think the error is more likely to be found in these structure checks.
Since only IBAN numbers from Switzerland and Lichtenstein are allowed, and both have the same structure, this additional checks could be simplified.

I have made a branch, which removes the iban dependency and implements the simplified IBAN checks directly into the library. You can install it using 
`npm i https://github.com/schoero/SwissQRBill.git#Fix-#215-iban-validation`

Please let me know if your IBAN passes the simplified checks.

**Comment by TheNetStriker** [comment]
I've just tested this, but I'm still getting an error:

```
/usr/local/bin/node ./RechnungMedlist.js
(node:89421) UnhandledPromiseRejectionWarning: Error: Creditor Reference requires the use of a conventional IBAN.
    at PDF._validateData (.../node_modules/swissqrbill/lib/swissqrbill.js:416:27)
    at new PDF (.../node_modules/swissqrbill/lib/swissqrbill.js:47:14)
    at new PDF (.../node_modules/swissqrbill/lib/node.js:42:13)
    at .../my.js:270:21
    at new Promise (<anonymous>)
    at generateQrInovice (.../my.js:269:12)
    at /my.js:317:17

```

**Comment by schoero** [comment]
Thank you for testing.
I just realized that I messed up the error messages there. The correct error message would be `QR-Reference requires the use of a QR-IBAN (and vice versa).`

I'm guessing you are trying to use a 27-digit QR-Reference in conjunction with your normal IBAN?

This is not allowed according the [specifications](https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf) (page 32, "Customer references"). If you want to continue using a 27-digit reference number (as in the orange ISR), you must request a QR-IBAN from your bank.
